subprojects/core/src/main/java/org/gradle/api/internal/tasks/cache/diagnostics/TaskOutputCacheEvent.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.cache.diagnostics;
+
+import org.gradle.api.Task;
+
+public abstract class TaskOutputCacheEvent {
+    final String taskName;
+    final String taskClass;
```

Comment:
Let's use `private` fields and getters here. We might want to mock these guys later on or something.